### PR TITLE
Add signed data under canTransfer check

### PR DESCRIPTION
**Pull request by satyamakgec**
`_to` address can be whitelisted and checked for the transfer.

### CODE REVIEW

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol [hunk 1]
```diff
@@ -480,41 +534,71 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
...
 
         bytes32 hash = keccak256(
             abi.encodePacked(this, _investor, _canSendAfter, _canReceiveAfter, _expiryTime, _validFrom, _validTo, _nonce)
         );
 
-        if (_checkSig(hash, _signature, _nonce)) {
+        (isVerifiedSig, signer) = _checkSigView(hash, _signature, _nonce);
+        if (isVerifiedSig) {
+            if (_isReadOnlyCall) {
+                for (uint256 i = 0; i < _investor.length; i++) {
```
Comment (adamdossa) [inline contracts/modules/TransferManager/GTM/GeneralTransferManager.sol]: Doesn't this mean that if the if statements fail in L571 or L575 the function will return (true, ...) even if `_isReadOnlyCall == true`? Not sure why we're only doing the uint64 checks when `_isReadOnlyCall == true`.

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol [hunk 2]
```diff
@@ -134,34 +134,85 @@ contract GeneralTransferManager is GeneralTransferManagerStorage, TransferManage
...
+                    bool success, 
+                    address[] memory investor,
+                    ,
+                    uint256[] memory canRecieveAfter,
+                    uint256[] memory expiryTime
+                ) = _processTransferSignatureView(data, validFrom, validTo, nonce);
+                if (success) {
+                    for (nonce = 0; nonce < investor.length; nonce++) {
+                        // Searching the _to token holder to get the required details for _verifyTransfer validation 
+                        if (investor[nonce] == _to)
```
Comment (adamdossa) [inline contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:171]: This logic doesn't quite match the `transferWithData` version as it is possible the signed data is relevant for the `from` address which wouldn't be picked up here I think.

**Review by adamdossa — CHANGES_REQUESTED** [review]
@satyamakgec it looks like there is a difference in behaviour in the GTM between `canTransferWithData` and `transferWithData`

File: contracts/modules/TransferManager/GTM/GeneralTransferManager.sol (hunk 2 above)
Comment (satyamakgec) [reply contracts/modules/TransferManager/GTM/GeneralTransferManager.sol:171]: Do you think `from` address will come under into the check, I am expecting `from` does posses the balance it means It is already whitelisted with the ST?

**Review by adamdossa — APPROVED** [review]

**Pull request merged**